File: pythonProject/temp/SlowUnittests.py
import csv
import os

import pytest
from App.Calculator import Calculator

path_to_file =os.getcwd() + ("\..\Resources\slow_unit_test_data.csv")
path_to_file2 =os.getcwd() + ("\..\Resources\slow_unit_test_data_2.csv")


def test_1_2_3():
    c = Calculator()
    c.press("1")
    c.press("2")
    c.press("3")
    assert c.getDisplay() == "123"


def get_test_data():
    with open(path_to_file, 'r') as f:
        reader = csv.reader(f)
        data = {tuple(row) for row in reader}
    return data

@pytest.mark.parametrize("input, output", get_test_data())
def test_multiple_vals(input, output):
    c = Calculator()
    for ch in input:
        c.press(ch)
    assert c.getDisplay() == output

def get_test_data2():
    with open(path_to_file2, 'r') as f:
        reader = csv.reader(f)
        data = {tuple(row) for row in reader}
    return data

@pytest.mark.parametrize("input, output", get_test_data2())
def test_multiple_vals_2(input, output):
    c = Calculator()
    c.pressAll(input)
    assert c.getDisplay() == output

File: CSharp/DirtyTestsWorkshop/Exercise/Temp/SlowUnitTests.cs
using CsvHelper;
using DirtyTestsWorkshop.App;
using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.Resources;
using NUnit.Framework;
using System.Formats.Asn1;
using System.Globalization;

namespace DirtyTestsWorkshop.Exercise.Temp
{
    [TestFixture]
    public class SlowUnitTests
    {
        private const string path_to_file = @"Exercise\Resources\slow_unit_test_data.csv";
        private const string path_to_file2 = @"Exercise\Resources\slow_unit_test_data_2.csv";

        [Test]
        public void test_1_2_3()
        {
            Calculator c = new Calculator();
            c.Press("1");
            c.Press("2");
            c.Press("3");
            Assert.AreEqual(c.GetDisplay(), "123");
        }


        [Test, TestCaseSource("GetTestData")]
        public void test_multiple_vals(string input, string output)
        {
            Calculator c = new Calculator();
            foreach (char ch in input)
            {
                c.Press(ch.ToString());
            }
            Assert.AreEqual(c.GetDisplay(), output);
        }

        [Test, TestCaseSource("GetTestData2")]
        public void test_multiple_vals_2(string input, string output)

        {
            Calculator c = new Calculator();
            c.PressAll(input);
            Assert.AreEqual(c.GetDisplay(), output);
        }

        private static IEnumerable<string[]> GetTestData()
        {
            using (var reader = new StreamReader(path_to_file))
            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
            {
                while (csv.Read())
                {
                    yield return new string[] { csv[0], csv[1] };

                }
            }
        }
        private static IEnumerable<string[]> GetTestData2()
        {
            using (var reader = new StreamReader(path_to_file2))
            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
            {
                while (csv.Read())
                {
                    yield return new string[] { csv[0], csv[1] };

                }
            }
        }
    }
}
